core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -307,6 +309,20 @@ public void registerSuggester(String key, Suggester<?> suggester) {
     }
 
     /**
+     * Lookup a {@link ScoreFunctionParser} by its name as used in XContent.
+     */
+    public ScoreFunctionParser<?> lookupScoreFunctionParser(String name, ParseFieldMatcher matcher) {
+        Tuple<ParseField, ScoreFunctionParser<?>> parser = scoreFunctionParsers.get(name);
+        if (parser == null) {
+            return null;
+        }
+        if (matcher.match(name, parser.v1())) {
+            return parser.v2();
+        }
+        return null;
```

Comment:
I think the other way around - if it were a private or protected method I'd have just pushed the error handling into it. As it is public it feels more normal to return null like a java collection. I can make it private I guess.